site/docs/v1/tech/lambdas/openid-connect-response-reconcile.adoc

```diff
@@ -26,25 +26,39 @@ This lambda must contain a function named `reconcile` that takes at least three
 * `user` - the FusionAuth User object
 * `registration` - the FusionAuth UserRegistration object
 * `jwt` - the JSON payload returned from the OpenID Connect Userinfo endpoint
-* `id_token` - the JSON payload returned in the `id_token` when available. This parameter may not always be provided.
+* `id_token` - the JSON payload returned in the `id_token` when available. This parameter may not be provided and will be `undefined` in that case.
 
-The two FusionAuth objects are well documented here in the link:/docs/v1/tech/apis/users/[User API] and link:/docs/v1/tech/apis/registrations/[Registration API] documentation. The JWT object that contains the payload from the Userinfo endpoint may contain well known OpenID Connect registered claims as well as any custom claims defined by the identity provider. The `id_token` will be provided to the Lambda when it is returned by the Identity Provider and it has been signed using the `client_secret` and an HMAC algorithm. If the `id_token` is signed using an asymmetric key-pair it will not be available to the Lambda.
+The two FusionAuth objects are well documented in the link:/docs/v1/tech/apis/users/[User API] and link:/docs/v1/tech/apis/registrations/[Registration API] documentation. 
 
-== Assigning the lambda
+The JWT object that contains the payload from the Userinfo endpoint. It may contain well known OpenID Connect registered claims as well as any custom claims defined by the Identity Provider. 
+
+The `id_token` will be provided to the Lambda only:
+
+* when it is returned by the Identity Provider and 
+* it has been signed using the `client_secret` and an HMAC algorithm
+
+In particular, if the `id_token` is signed using an asymmetric key-pair, it will not be available to the Lambda.
+
+== Assigning the Lambda
 
 Once a lambda is created, you may assign it to one or more OpenID Connect IdPs in the IdP configuration.
 
 Navigate to [breadcrumb]#Settings -> Identity Providers# and select your existing an OpenID Connect configuration or click [breadcrumb]#Add provider# and select OpenID Connect if it has not yet been configured.
 
-== Example lambda
+== Example Lambda
 
 include::docs/v1/tech/lambdas/_openid-connect-example-lambda.adoc[]
 
-=== Workarounds
+== Modifying Email and Username Claims
 
 If the JWT from the OIDC identity provider does not come back with an email claim you can add your own. This claim is `email` by default but may be changed with the `oauth2.emailClaim` as documented in the link:/docs/v1/tech/apis/identity-providers/openid-connect/[API docs].
 
-If the Userinfo response available to you in the lambda has unique user information, you can build a fake email address from it.
+[NOTE.note]
+====
+This capability is available beginning in version `1.31.0`. It was also available from `1.17.3` to `1.28.0`.
+====
```

Comment:
This is fine - I had removed the intermediate version stuff on purpose just to make it simpler. My thinking was that anyone running those versions is already all set up, and if it works, it works. 

If they are looking at the doc - for this feature because they are not yet using it, they'll just see it is available in 1.30.0 and upgrade. Getting them to the most recent version is a win in any case. 

But if you'd prefer to keep it that is ok.

Reply:
Hmmm. Versioning is hard. But I do like to give folks info about major shifts, so would prefer to keep it in.